### PR TITLE
Added tests for HashFunctions.java

### PR DESCRIPTION
**Pull request by freddytuxworth**
Hi,
I've analysed your codebase and have noticed some gaps in the coverage of HashFunctions.java.

I've written a few tests for this class with the help of Diffblue Cover.

Hopefully these tests should help you detect regressions caused by future code changes. Please let me know if you'd be interested in having more tests written for other classes within this project, I would be happy to help.

### CODE REVIEW

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]
I would remove the comments from the tests, especially the `/* testedClasses` ones. 

Also, is this the PR title we're going to use from now on? The `(from Cover with manual additions)` seems clunky to me.

Also, remove any duplicate/near-duplicate tests. I don't think we use tests for hashes because we currently just return zero all the time for them anyway, actually.

File: src/test/java/com/flowpowered/math/test/HashFunctionsTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+import com.flowpowered.math.HashFunctions;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class HashFunctionsTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
```
Comment (petersear) [inline src/test/java/com/flowpowered/math/test/HashFunctionsTest.java]: It doesn't look like we need this, and the associated import statement for Rule, as there are no tests which throw exceptions here.

File: src/test/java/com/flowpowered/math/test/HashFunctionsTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    // Arrange
+    final float v = 1.5f;
+
+    // Act
+    final int retval = HashFunctions.fastCeil(v);
+
+    // Assert result
+    Assert.assertEquals(2, retval);
+  }  /* testedClasses: HashFunctions */
```
Comment (petersear) [inline src/test/java/com/flowpowered/math/test/HashFunctionsTest.java]: Remove "testedClasses" comment

**Review by petersear — CHANGES_REQUESTED** [review]

**Review by petersear — APPROVED** [review]

**Comment by sudo-jaa** [comment]
Almost good to go. @freddytuxworth, do you know how to squash commits?

**Comment by petersear** [comment]
Something I forgot to mention - it's worth including the proposed text for the PR so that can be reviewed as well. In practice, it's standard text but still worth checking.
Something else...the commits will need squashing to one and I'd suggest bringing the commit message in line with the PR title as suggested by John ie. leave out "from Cover with manual additions" (btw per James Wilson, our product should be referred to externally as "Diffblue Cover", although inevitably we will refer to it as "Cover" internally.

**Review by paulkennethkent — COMMENTED** [review]
Looks good, I would say that the only thing that is missing is very minor. In our commit messages, we usually go for a multiline message and include the following:

```
<commit message> 
These tests were written using Diffblue Cover.
Dependencies required to run the tests have also been added.  
```
Only add in the dependencies line if you've made changes to the pom.xml

**Comment by sudo-jaa** [comment]
@freddytuxworth Once you've implemented above changes, let me know and we'll squash back down and get to submitting the PR proper.

**Review by sudo-jaa — APPROVED** [review]

**Comment by freddytuxworth** [comment]
I've added "These tests were written using Diffblue Cover" to the commit message.

**Review by JohnLBergqvist — APPROVED** [review]

File: src/test/java/com/flowpowered/math/test/HashFunctionsTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    final Object object = null;
+    final int retval = HashFunctions.hash(object);
+    Assert.assertEquals(0, retval);
+
+  }
+
+  @Test
+  public void hashIntInputZeroOutputZero() {
+
+    final int value = 0;
```
Comment (peterschrammel) [inline src/test/java/com/flowpowered/math/test/HashFunctionsTest.java:59]: Use something other than 0.

**Pull request closed without merging**